src/configurePipeline/services/source/sourceRepoService.ts

```diff
@@ -0,0 +1,172 @@
+import * as vscode from 'vscode';
+
+import * as fs from 'fs';
+import * as git from "simple-git/promise";
+import * as Mustache from "mustache";
+
+import { GitRepoDetails, SourceProviderType } from '../../model/Common';
+import { AzureDevOpsService } from "../azureDevOpsService";
+import Q = require('q');
+
+export class SourceRepoService {
+    private gitReference: git.SimpleGit;
+    constructor() {
+
+    }
+
+    /**
+     * @returns repoObject: localPath, provider, repoName, repoId, repoUrl, currentRemote, branch
+     */
+    public getSourceRepo(azureDevOps: AzureDevOpsService) {
```

Comment:
I think we should not have azureDevOps parameter here, see comment below getGitRepoDetails()